### ISSUE TITLE
[Bug] Damage determination may override shields in some cases

### ISSUE THREAD

**Issue by WorldlineChanger** [Melledy/Grasscutter#980]
**Describe the bug**

In situations where enemies around the character are attacking with high frequency, the character may be hurted even while under shield protection.
For example, with 10-20 enemies around, it is still possible for the character to take damage from the shield even if the shield is not broken (normal attack series, not involving energy-related attacks in the recent versions of the game)

**Which branch did you use?**

Grasscutter version: 1.1.0-dev

**Comment by Theosapira** [comment]
Did you use burst? Because you take dmg if you use burst, even if you have shield active.

**Comment by Birdulon** [comment]
This happens on official servers too.

**Comment by WorldlineChanger** [comment]
> Did you use burst? Because you take dmg if you use burst, even if you have shield active.

That's true, thanks a lot.

**Comment by WorldlineChanger** [comment]
So seems like invincible frame doesn't work on Grasscutter.

**Comment by Theosapira** [comment]
> So seems like invincible frame doesn't work on Grasscutter.

Yeah, so probably what's happening is that shields "turn off" during bursts to not decrease their durability but because there are no iframes on ps you take dmg, unlike official servers. So the problem is the iframes.

**Comment by Birdulon** [comment]
> > So seems like invincible frame doesn't work on Grasscutter.
> 
> Yeah, so probably what's happening is that shields "turn off" during bursts to not decrease their durability but because there are no iframes on ps you take dmg, unlike official servers. So the problem is the iframes.

Shields take damage during bursts on official servers.

**Comment by Theosapira** [comment]
> > > So seems like invincible frame doesn't work on Grasscutter.
> > 
> > 
> > Yeah, so probably what's happening is that shields "turn off" during bursts to not decrease their durability but because there are no iframes on ps you take dmg, unlike official servers. So the problem is the iframes.
> 
> Shields take damage during bursts on official servers.

They were but I think there was an update that they changed it to not take dmg
Edit: Yep just tested it, shields don't take dmg during burst on official servers
Edit2: Found it, it was on 2.1
![image](https://user-images.githubusercontent.com/105167228/169252939-f39afeeb-035c-4832-af0f-82f97d69f766.png)

**Comment by Birdulon** [comment]
2.8 update: On official servers, just took 11k damage from explosive barrels with a shield up.

**Comment by Theosapira** [comment]
> 2.8 update: On official servers, just took 11k damage from explosive barrels with a shield up.

Were these barrels from the inazuma bounty, which deal increased dmg?

Either the barrels dealt more dmg than your shield's durability and your shield broke or if you used zhongli's hold E to activate the shield too close to the barrels, the shockwave of his E happens before the shield (not at the same time), so you took the damage before the shield actually activates.

**Comment by Birdulon** [comment]
> Were these barrels from the inazuma bounty, which deal increased dmg?

Mondstadt

> Either the barrels dealt more dmg than your shield's durability and your shield broke

The shield remained for the full duration and blocked subsequent instances of damage.

> zhongli's hold E

Yes, I make a point of always doing that before running into range of the barrels since I've had that happen before.

I also tend to accumulate chip damage on Raiden fairly often, likely from electro-charged. It's pretty noticeable and memorable when you have full HP OCD and run teams without healers. Shields are just wonky in this game.
